File: src/main/java/yunas/Controller.java
package yunas;

import yunas.Context;

import java.util.function.Function;

/**
 * yunas.Controller interface for java.
 */
public interface Controller extends Function<Context,Object> {

    public Object action(Context context);

    @Override
    default Object apply(Context context) {
        return action(context);

    }
}

File: src/main/kotlin/yunas/Controller.kt
package yunas

import java.util.function.Function

/**
 * yunas.Controller interface for java.
 */
@FunctionalInterface
interface Controller : Function<Context, Any> {

    fun action(context: Context): Any

    override fun apply(context: Context): Any {
        return action(context)

    }
}
